### PR TITLE
build(deps): bump forbole/juno to v3.0.0

### LINKED ISSUE #282: Implement an upgrade command
Short: implement an upgrade command that allows to seamlessly update from one BDjuno version to the other without husttle

Long: TBD

### LINKED ISSUE #306: GetMessagesByAddress optimisation
Currently when the Tx table gets too big `GetMessagesByAddress` will stop working correctly. This might be an issue with missing indexes.

Below is the query i'm currently using:
```
query GetMessagesByAddress($address: _text, $limit: bigint = 50, $offset: bigint = 0 $types: _text = "{}") {
  messagesByAddress: messages_by_address(args: {addresses: $address, types: $types, limit: $limit, offset: $offset}) {
    transaction {
      height
      hash
      success
      messages
      logs
      block {
        height
        timestamp
      }
    }
  }
}

```

Even if the limit is set to 10 the data will not return. 
This is currently happening on Band and Sifchain

### LINKED ISSUE #346: `bdjuno parse-genesis` cmd requires access to node
Can't parse genesis file without access to the node. This should be changed so we can process the genesis file before the chain starts

### PR DESCRIPTION
**Pull request by MonikaCat**
## Description
This PR updated Juno to `v3.0.0`. 

Closes: #282
Closes: #346
Closes: #306

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [x] targeted the correct branch
- [x] provided a link to the relevant issue or specification
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,13 +4,12 @@ go 1.17
 
 require (
 	github.com/cosmos/cosmos-sdk v0.42.9
-	github.com/forbole/juno/v2 v2.0.0-20220223115557-e4ed06a008de
+	github.com/forbole/juno/v3 v3.0.0-20220315104134-6cea725e4656
```
Comment (huichiaotsou) [inline go.mod]: should we use the latest juno version? 
```suggestion
	github.com/forbole/juno/v3 v3.0.0-20220324090024-29c52ffaa060
```

**Review by rmontagnin — APPROVED** [review]
utACK

**Comment by huichiaotsou** [comment]
should we modify the schema in 00-cosmos.sql to support table partitioning?

**Comment by rmontagnin** [comment]
> should we modify the schema in 00-cosmos.sql to support table partitioning?

Yes we should update all the necessary SQL schemes that have been updated inside Juno 3.0.0

**Review by huichiaotsou — APPROVED** [review]
tested Ack

**Pull request merged**